mloop_config.py

```diff
@@ -1,110 +1,135 @@
 import os
 import json
 import configparser
+import tomli
 import logging
+from collections import namedtuple
 
 
-def get(config_path=None):
+MloopParam = namedtuple("MloopParam", ["name", "min", "max", "start"])
+RunmanagerGlobal = namedtuple("RunmanagerGlobal", ["name", "expr", "args"])
+
+def is_global_active(config, group, category)
+    """
+    We check to see if the requeted global has been activated or not
+    """
+
+    if group in config["MLOOP"]["groups"]:
+        if ("enable" not in config[category][group]) or config[category][group]:
+            return True
+    
+    return False
+
+def prepare_globals(global_list, params_val_dict):
+    globals_to_set = {}
+    for g in global_list:
+        target = g.name
+        args = [ params_val_dict[arg] for arg in g.args ]
+
+        assert args
+
+        if g.expr:
+            val = eval(g.expr)(*args)
+        else:
+            val = args[0]
+
+        globals_to_set[target] = val
+
+    return globals_to_set
+
+
+def get(config_paths=None):
     """Creates config file from specified file, or
     creates one locally with default values.
     """
 
     # Default to local directory and default name
-    if not config_path:
+    if not config_paths:
+        config_paths = []
         folder = os.path.dirname(__file__)
-        config_path = os.path.join(folder, "mloop_config.ini")
-
-    # Instantiate RawConfigParser with case sensitive option names
-    config = configparser.RawConfigParser()
-    config.optionxform = str
-
-    # Check if file exists and initialise with defaults if it does not
-    if os.path.isfile(config_path):
-        # Retrieve configuration parameters
-        config.read(config_path)
+        config_paths.append(os.path.join(folder, "mloop_config.toml"))
+        config_paths.append(os.path.join(folder, "mloop_config.ini"))

```

Comment:
Does `tomli.load()` work with `.ini` files? If not then I don't think we should include `mloop_config.ini` here

Reply:
In the newest changes I made (see my final comment) I have totally ditched the .ini files.  It is silly to support two standards.